test/exchange-connector/resourceLoader/fatRamlResourceLoader.test.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+"use strict";
+
+import { FatRamlResourceLoader } from "../../../src";
+
+import { expect, default as chai } from "chai";
+import chaiAsPromised from "chai-as-promised";
+import sinon from "sinon";
+import amf from "amf-client-js";
+
+before(() => {
+  chai.should();
+  chai.use(chaiAsPromised);
+});
+
+describe("Commerce standards resource loader fsAdapter tests", () => {
+  it("Throws error on undefined resource absolute path", () => {
+    const ccStandardsLoader = new FatRamlResourceLoader("file:///rootFolder");
+
+    return expect(() =>
+      ccStandardsLoader.fsAdapter.readFileSync(undefined)
+    ).to.throw();
+  });
+});
+
+describe("Commerce standards resource loader create tests", () => {
+  it("Throws error on undefined root path", () => {
+    return expect(() => new FatRamlResourceLoader(undefined)).to.throw(
+      "Cannot read property 'split' of undefined"
+    );
+  });
+
+  it("Throws error on null root path", () => {
+    return expect(() => new FatRamlResourceLoader(null)).to.throw(
+      "Cannot read property 'split' of null"
+    );
+  });
+
+  it("Throws error on root path without 'file://' prefix", () => {
```

Comment:
could use some more edge tests.

Like

```
file://///something
```

```
FILE://
```
etc

Write tests with the intention of breaking your stuff.